test/e2e/set/scyllacluster/scyllacluster_alternator.go

```diff
@@ -0,0 +1,225 @@
+
+	g.It("should setup Alternator API when enabled", func() {
+		ctx, cancel := context.WithTimeout(context.Background(), testTimeout)
+		defer cancel()
+
+		sc := f.GetDefaultScyllaCluster()
+		o.Expect(sc.Spec.Datacenter.Racks).NotTo(o.BeEmpty())
+		sc.Spec.Datacenter.Racks = sc.Spec.Datacenter.Racks[:1]
+		sc.Spec.Datacenter.Racks[0].Members = 1
+		sc.Spec.Alternator = &scyllav1.AlternatorSpec{}
+
+		framework.By("Creating a ScyllaCluster with 1 member")
+		sc, err := f.ScyllaClient().ScyllaV1().ScyllaClusters(f.Namespace()).Create(ctx, sc, metav1.CreateOptions{})
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		framework.By("Waiting for the ScyllaCluster to rollout (RV=%s)", sc.ResourceVersion)
+		waitCtx1, waitCtx1Cancel := utils.ContextForRollout(ctx, sc)
+		defer waitCtx1Cancel()
+		sc, err = controllerhelpers.WaitForScyllaClusterState(waitCtx1, f.ScyllaClient().ScyllaV1().ScyllaClusters(sc.Namespace), sc.Name, controllerhelpers.WaitForStateOptions{}, utils.IsScyllaClusterRolledOut)
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		verifyScyllaCluster(ctx, f.KubeClient(), sc)
+
+		svcIP, err := utils.GetIdentityServiceIP(ctx, f.KubeClient().CoreV1(), sc)
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		framework.By("Setting up Alternator credentials over CQL")
+
+		cqlConfig := gocql.NewCluster(svcIP)
+		cqlConfig.Consistency = gocql.All
+		cqlSession, err := cqlConfig.CreateSession()
+		o.Expect(err).NotTo(o.HaveOccurred())
+		defer cqlSession.Close()
+
+		awsCredentials := aws.Credentials{
+			AccessKeyID:     "alternator",
+			SecretAccessKey: "salted-password-42",
+		}
+		wrongAWSCredentials := awsCredentials
+		wrongAWSCredentials.SecretAccessKey += "-wrong"
+
+		err = cqlSession.Query(
+			`INSERT INTO system_auth.roles (role, salted_hash) VALUES (?, ?)`,
+			awsCredentials.AccessKeyID, awsCredentials.SecretAccessKey,
+		).Exec()
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		framework.By("Verifying that Alternator is configured correctly")
+
+		alternatorServingCABundleConfigMap, err := f.KubeClient().CoreV1().ConfigMaps(f.Namespace()).Get(ctx, fmt.Sprintf("%s-alternator-local-serving-ca", sc.Name), metav1.GetOptions{})
+		o.Expect(err).NotTo(o.HaveOccurred())
+		alternatorServingCACerts, _ := verification.VerifyAndParseCABundle(alternatorServingCABundleConfigMap)
+		o.Expect(alternatorServingCACerts).To(o.HaveLen(1))
+		alternatorServingCAPool := x509.NewCertPool()
+		alternatorServingCAPool.AddCert(alternatorServingCACerts[0])
+
+		httClient := awshttp.NewBuildableClient().WithTransportOptions(func(transport *http.Transport) {
+			transport.TLSClientConfig = &tls.Config{
+				RootCAs: alternatorServingCAPool,
+			}
+		})
+		config := aws.Config{
+			HTTPClient: httClient,
+			Credentials: credentials.StaticCredentialsProvider{
+				Value: awsCredentials,
+			},
+			RetryMaxAttempts: 1,
+		}
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		for _, alternatorURL := range []string{
+			fmt.Sprintf("https://%s:8043", svcIP),
```

Comment:
loop over single element slice? Looks like something is missing here, probably the identity service domain

Reply:
added a version with the DNS name that can still be run from our laptops